intro-to-automated-tests/intro-to-pytest.md

```diff
@@ -0,0 +1,219 @@
+# Intro to pytest
+
+## Learning Goals
+
+- Use automated unit tests to verify code correctness
+
+## Introduction
+
+When we understand that tests can verify code correctness, let's learn one way to use and write tests. The actual tools needed to use tests will depend on the language and technologies the project is in. Often, software teams have already decided their testing framework. In this curriculum, we've decided to test our Python code with `pytest`.
+
+## Vocabulary and Synonyms
+
+| Vocab | Definition | Synonyms | How to Use in a Sentence
+| --- | --- | --- | ---
+Assertion | A statement that somethng is true |
+Assert | Stating that something is true | expect |
+
+## Installing `pytest`
+
+[`pytest`](https://docs.pytest.org/en/stable/) is one of many testing tools for the Python language.
+
+We can install (and upgrade) the `pytest` package with this command:
+
+```bash
+pip3 install -U pytest
+```
+
+Callout in Red: Recall that we use `pip3` to designate `pip` for Python3.
+
+## Reworking Our Project Structure
+
+When we use `pytest`, we will be creating more files. In order to anticipate more files, some future projects and assignments will follow this folder structure:
+
+```bash
+.
+├── projectname
+│   ├── __init__.py
+│   └── somefile.py
+└── tests
+    ├── __init__.py
+    ├── context.py
+    └── test_somefile.py
+```
+
+Take care to notice the following:
+1. In our project directory, there are 0 files, and 2 folders: `projectname` and `tests`.
+    - We will replace the folder name `projectname` with our project's name
+1. In our `projectname` folder, there is a file named `somefile.py`
+    - This represents any Python files that include the functions we're writing
+1. In our `tests` folder, there is a file named `test_somefile.py`
+    - This represents a test file that is responsible for testing `projectname/somefile.py`
+1. There are some other files: `__init__.py` and `context.py`
+    - We will not focus on these files, but feel free to read through them
```

Comment:
Yup!

Reply:
Let's keep an eye on this-- I technically haven't written these repos yet, so there's a chance this could change. I'm hoping that it won't be a huge lift to update this if needed